platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -18,61 +18,86 @@ components within the platform.
-it's time to begin work on adopting a new address book (scope beyond this proposal), it will create a `Roster`
-object with information from the Address Book and pass it to the platform. The mechanism for doing so is detailed below.
+consensus and block signing. The Roster, a subset of the address book, will provide the data structure and API for this mechanism.
+This proposal provides a specification for the behavior of rosters from creation to terminal state.
+A roster reaches a terminal state when it is removed from the state store.
+The details of how the roster is created, validated, stored, adopted and removed are outlined in this proposal.
 
 ## What this proposal does not address
 
+1. Details relating to what components will call the new APIs to store and retrieve rosters. This belongs in the other TSS proposals.
+2. How devops will manage the lifecycles of artifacts used in creating the roster.
+
 ## Requirements
 
-- Immutability: The roster hash will be a critical piece of data. Rosters must be immutable to protect the integrity
-  of its computed hash.
-- Efficient Storage: Roster data should be stored efficiently in the state to minimize overhead and unnecessary data
-  copying.
-- Clear API: A well-defined API should be provided for the App to create and submit candidate rosters to the platform,
-  and for the platform to manage the lifecycle of submitted rosters.
-- Components update: Platform components that currently rely on the use of the Address book (such as Reconnect, Network,
-  Event Validation etc.) must be adapted to use the new `Roster` object instead.
-- Address book paradigm discontinued within the platform codebase in favor of `Roster`s
+1. Immutability: The roster hash will be a critical piece of data. Rosters must be immutable to protect the integrity of its computed hash.
+2. Efficient Storage: Roster data should be stored efficiently in the state to minimize overhead and unnecessary data copying.
+3. Clear API: A well-defined API should be provided to create and submit candidate rosters to the platform, and for the platform to manage the lifecycle of submitted rosters.
+4. Components update: Platform components that currently rely on the use of the Address book (such as Reconnect, Network, Event Validation etc.) must be adapted to use the new `Roster` object instead.
+5. Address book paradigm discontinued within the platform codebase in favor of `Roster`s.
+6. There must be a well-laid-out mechanism for the start-up code to arrive at the correct roster to use in starting the network.
 
 ## Architecture
 
-###### Roster Lifecycle
+### Roster Types.
+
+This proposal introduces four types of `Roster`s - candidate `Roster`, active `Roster`, genesis `Roster`, and override `Roster`.
 
-![](TSS%20Roster%20Lifecycle.drawio.svg)
+1. A Candidate `Roster` is a Roster that is currently being considered for adoption to become the active `Roster`.
+2. An Active `Roster` is a Roster currently being used for signing of blocks.
```

Comment:
I think the dilema when I was writing this is ... do we frame this in the end goal context vs today's? I went for the end goal but I'm not sure. If we do use today's framing, it will have to carry the same definition as a the active address book which itself too may be confusing.

Reply:
Will "An Active Roster is the roster currently being used to determine consensus" work?

Reply:
Yes, this latter phrasing looks perfect to me. Thanks!